torch_xla/csrc/xla_sharding_util.cpp

```diff
@@ -572,63 +567,63 @@ void ShardingUtil::PrepareOutputShardingPropagation(
     }
   }
 
-  // Output parameter sharding annotations, defaults to REPLICATED(0) if unset.
+  // Output parameter sharding annotations, defaults to REPLICATED(0) if
+  // unset.
   if (output_shardings.empty()) {
     // Initializes with default sharding type, REPLCIATED.
-    output_shardings.resize(indices.size());
+    output_shardings.resize(num_outputs);
```

Comment:
we can remove `num_outputs` arg from the function, and use `sharding_specs.size()` instead. But you will remove this function entirely in a separate PR, right?